ecs/compatibility.go

```diff
@@ -135,3 +135,12 @@ func (c *fargateCompatibilityChecker) CheckLoggingDriver(config *types.LoggingCo
 		c.Unsupported("services.logging.driver %s is not supported", config.Driver)
 	}
 }
+
+func (c *fargateCompatibilityChecker) CheckUlimits(service *types.ServiceConfig) {
```

Comment:
I don't get who calls these CheckXXX methods, is it AWS using reflection ? does CheckULimits implement an AWS interface ?

Reply:
compose-go does: https://github.com/compose-spec/compose-go/blob/a70cd5945c2555b0d96d74cdecd67542331c9bdb/compatibility/checker.go#L364

by defining this (I'd say "override" if golang allowed it) we customize the default mechanism, which is just opt-in for supported attributes, so we can control allowed values